src/System.Private.CoreLib/shared/System/Collections/Concurrent/ConcurrentQueue.cs

```diff
@@ -276,26 +276,29 @@ public int Count
         {
             get
             {
-                ConcurrentQueueSegment<T> head, tail;
-                int headHead, headTail, tailHead, tailTail;
+                // Take the cross-segment lock to prevent _head and _tail from changing while we're
```

Comment:
You're not taking the lock at this point but the comment says you are.

Reply:
Oops, leftover from a refactoring.  Will fix.